### PR TITLE
[venstar] Add semantic tags to channels

### PR DESCRIPTION
**Pull request by digitaldan**
Signed-off-by: Dan Cunningham <dan@digitaldan.com>

### CODE REVIEW

**Comment by hww3** [comment]
LGTM!

**Review by hww3 — APPROVED** [review]
Looks good!

File: bundles/org.openhab.binding.venstarthermostat/src/main/resources/OH-INF/thing/thing-types.xml [hunk 1]
```diff
@@ -114,6 +114,9 @@
 		<item-type>String</item-type>
 		<label>System Mode</label>
 		<description>Current System Operating Mode</description>
+		<tags>
+			<tag>Control</tag>
```
Comment (lsiepel) [inline bundles/org.openhab.binding.venstarthermostat/src/main/resources/OH-INF/thing/thing-types.xml:118]: Might conflict with rule 2: https://github.com/openhab/openhab-addons/issues/12262

**Comment by digitaldan** [comment]
> Might conflict with rule 2: https://github.com/openhab/openhab-addons/issues/12262

Hmm, i would agree that it does as well.  But i am also confused then what "control" is meant with semantic tags?  It would seem reasonable to me that a channel the handles the "control" of a thing, would be tagged as such?  I am very much not clear then how such tags should be used and what benefit there is / is not with such a tag? @lolodomo maybe you can explain ?

**Comment by lsiepel** [comment]
> o me that a channel the handles the "control" of a thing, would be tagged as such? I am very much not clear then how such tags should be used and what benefit there is /

I think the tag alone is not specific enough and it also needs a tag _what_ it controls, like a receiver, hvac etc. But i must agree that i find the semenatic tags a bit of a vague at some point. I really hope someone can create a set of rules and add it to the   xml definition. It starts with the rules tough ;-)

**Comment by digitaldan** [comment]
@lolodomo any feedback?  Thanks!

File: bundles/org.openhab.binding.venstarthermostat/src/main/resources/OH-INF/thing/thing-types.xml [hunk 2]
```diff
@@ -284,6 +287,9 @@
 		<item-type>String</item-type>
 		<label>System State</label>
 		<description>Current System Operating State</description>
+		<tags>
+			<tag>Status</tag>
+		</tags>
```
Comment (lolodomo) [inline bundles/org.openhab.binding.venstarthermostat/src/main/resources/OH-INF/thing/thing-types.xml:292]: This one is unexpected.

**Review by lolodomo — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.venstarthermostat/src/main/resources/OH-INF/thing/thing-types.xml (hunk 1 above)
Comment (lolodomo) [reply bundles/org.openhab.binding.venstarthermostat/src/main/resources/OH-INF/thing/thing-types.xml:118]: Yes, this one is unexpected.

**Comment by lolodomo** [comment]
Only the tags you set for temperature, setpoint and humidity look good to me, considering the discussions about semantic tags we have in the past (in particular with @kaikreuzer) leading to few rules (#12262).

**Comment by digitaldan** [comment]
Thanks @lolodomo , i have read that list, but i'm still not clear about this:

>It is not allowed to set only "Control" or "Switch" tag on a writable channel type. Also not expected to set "Status" tag on all read-only channel types.

I guess i don't understand "why" its not allowed?  And what would be allowed for a "Control" tag?

**Comment by lolodomo** [comment]
> > It is not allowed to set only "Control" or "Switch" tag on a writable channel type. Also not expected to set "Status" tag on all read-only channel types.
> 
> I guess i don't understand "why" its not allowed? And what would be allowed for a "Control" tag?

You have to mention what is controlled, for example "Control" + "Light".

**Comment by digitaldan** [comment]
So in the case of HVAC equipment, the channel which controls the primary operational mode (HEAT,COOL,AUTO,OFF) would have what additional tag?

**Comment by lsiepel** [comment]
> So in the case of HVAC equipment, the channel which controls the primary operational mode (HEAT,COOL,AUTO,OFF) would have what additional tag?

Control + HVAC tags  I suppose

More equipment here:  https://github.com/openhab/openhab-core/blob/main/bundles/org.openhab.core.semantics/model/SemanticTags.csv

**Comment by lolodomo** [comment]
A tag for Point + a tag for Property.
I am not sure what property is appropriate in that case. Maybe "Energy" ?
"Control" + "Energy" ?

**Comment by lsiepel** [comment]
> hich controls the primary op

i would suggest to use: Control + Temperature

**Comment by lsiepel** [comment]
@digitaldan do you need more information or can you proceed with this PR?

**Comment by lsiepel** [comment]
I close this PR due to inactivity. Personally i think that pre-set tags in general can be added to many more bindings, so i hope that at some point this will be picked up again.

However, source is not gone. When you (or someone else) is able to proceed with this PR it can be re-opened, or feel free to create a new PR. I would be happy to review and merge it.

**Pull request closed without merging**